cumulus/polkadot-parachain/src/service.rs

```diff
@@ -251,6 +235,11 @@ where
 	let client = params.client.clone();
 	let backend = params.backend.clone();
 
+	let info = backend.blockchain().info();
+	if !client.runtime_api().has_basic_apis(info.genesis_hash) {
```

Comment:
I am not sure if this works atm, some of these system chains were initially run as `Shell` and might not have the aura api in the initial blocks afaik. 

Alternatively, the changes in https://github.com/paritytech/polkadot-sdk/tree/kiz-omni-node is one way to keep this crate mostly unchanged, and only enable the omni-node when an unknonw chain-spec is given.

We could subsume that branch here.

Reply:
Specifically, see how I have replaced `enum Runtime::Default` with `Runtime::Omni`, and all new codes are constrained to this code path.

Reply:
Tbh, solely because of these legacy decisions in system chains and then requiring special code such as `WaitForAuraConsensus`, I am in favour of not touching `polkadot-parachain`, and starting a fresh new omni-node crate, keep `polkadot-parachain` unchanged, and drop-in replace it once done. @bkchr iirc you had some reservations about this, wdyt?

Reply:
Good point. You're right. It wasn't working when syncing system parachains. Removed the `has_basic_apis()` check and also the check for `AuraUnincludedSegmentApi`.

Reply:
> I am not sure if this works atm, some of these system chains were initially run as `Shell` and might not have the aura api in the initial blocks afaik.

But that only applies to asset hub runtimes, which is why they use the separate `start_asset_hub_lookahead_node()` entry point. So, might it still be okay to keep that check in the general `start_generic_aura_lookahead_node()`?

Reply:
The problem is that we would have to perform this check somewhere after `new_partial()` was called. And I don't think there is a generic way to do this now just for `start_asset_hub_lookahead_node()`. I think it would be better to do it in a future PR together with some more refactoring.